doc/user-guide/template.md

```diff
@@ -294,3 +293,25 @@ sex = "FEMALE"
 %%sql
 SELECT * FROM one_sex
 ```
+
+## Disabling named parameters
+
+The named parameters option can be _Disabled_ using `%config SqlMagic.named_parameters = 2`
```

Comment:
why are we using numbers? they're not very informative?

it'd be better to use strings. i saw some comments about it but it's still unclear

Reply:
This change was made per @neelasha23's [comment here.](https://github.com/ploomber/jupysql/pull/980#discussion_r1455531317) The thought was users would confuse string values with boolean, ie. they would write `%config SqlMagic.named_parameters=false` instead of `%config SqlMagic.named_parameters="false"` so the integer values were more of a distinction. @edublancas I can change it based on whatever you and @neelasha23 decide

Reply:
this issue has been stuck for a while so let's simplify this so we can merge this asap.

currently, `%config SqlMagic.named_parameters=false` is the default value. if it detects that your query has `:variable`, it'll produce this error:

```
In [5]: %sql select * from penguins.csv limit :limit
Running query in 'duckdb://'
RuntimeError: (Your query contains named parameters (limit) but the named parameters feature is disabled. Enable it with: %config SqlMagic.named_parameters=True)
(sqlalchemy.exc.InvalidRequestError) A value is required for bind parameter 'limit'
[SQL: select * from penguins.csv limit $1]
[parameters: [{}]]
(Background on this error at: https://sqlalche.me/e/20/cd3x)
If you need help solving this issue, send us a message: https://ploomber.io/community
```

I did this to help users, and I didn't anticipate that this would cause issues. so, let's do this:

let's keep true/false as the allowed values. but false should now complete disable named parameters, effectively fixing: https://github.com/ploomber/jupysql/issues/972

let's not add any deprecation warnings, we'll just break the API. we try not to break the API that often but this is a niche feature that I don't think many users know about. since this will break the API, we'll have to change the current development version from `0.10.8dev` to `0.11.0dev`

questions @bryannho?